### PR TITLE
Star Rating Widget

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** new widget

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] For new widgets, `theme.variant()` is added to the root domnode
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

Adds a rate widget with customisable icons

### CODE REVIEW

File: src/rate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	/* The form name for this rate widget */
+	name?: string;
+	/* Flag to indicate if half stars should be used */
+	allowHalf?: boolean;
+	/* Flag to set the readonly state */
+	readOnly?: boolean;
+}
+
+export interface RateChildren {
+	label?: RenderResult;
```
Comment (maier49) [inline src/rate/index.tsx:30]: If we make a new parade release, jsdoc comments on these fields will be shown in a `children` table. Do we want to start adding them generally?

**Review by maier49 — APPROVED** [review]
Looks good to me. Just had one question about child interface comments going forward

File: src/rate/tests/Rate.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+import Rate from '../index';
+import RadioGroup from '../../radio-group';
+
+import * as fixedCss from '../rate.m.css';
+import * as css from '../../theme/default/rate.m.css';
+import * as baseCss from '../../common/styles/base.m.css';
+
+import { Icon } from '../../icon';
+
+const noop = () => {};
```
Comment (msssk) [inline src/rate/tests/Rate.spec.tsx:16]: this is so ubiquitous that it's exported by `common/tests/support/test-helpers`

File: src/rate/index.tsx (hunk 1 above)
Comment (tomdye) [reply src/rate/index.tsx:30]: good shout

**Pull request merged**